projects/wave-core/src/lib/operators/dialogs/data-repository/vector/vector-source-dataset.component.ts

```diff
@@ -96,6 +96,68 @@ export class VectorSourceDatasetComponent implements OnInit {
         });
         this.projectService.addLayer(l);
     }
+    // maybe refactor the following methods with central methods for finding min/max etc.
+    ngOnChanges(changes: SimpleChanges) {
+        for (const key in changes) {
+            if (changes.hasOwnProperty(key)) {
+                switch (key) {
+                    // check if there is any time-validity start/end data. If there is start/end data show the column of this data.
+                    // Also find a global Start/End (Min/Max) for the whole dataset.
+                    case 'dataset': {
+                        let globalDatasetMin = this.dataset.time_start;
+                        let globalDatasetMax = this.dataset.time_end;
+                        this.dataset.vectorLayer.forEach((element) => {
+                            for (const sourceVar in element) {
+                                if (element.hasOwnProperty(sourceVar)) {
+                                    if (sourceVar === 'time_start') {
```

Comment:
You know the type of `element`, so you can access the fields directly without needing to iterate over them, check field names, etc.

Reply:
Its the same for the checks in source-dataset.component.ts

Reply:
deleted the useless for-loop in 6da1d4c74b7a75a702074ebf5ef83c4bda092d4f